model/ArrayList.php

```diff
@@ -360,6 +360,7 @@ protected function parseSortColumn($column, $direction = null) {
 		}
 
 		// Parse column specification, considering possible ansi sql quoting
+		$column = str_replace('"', '', $column); // Remove quotes so sorts like '"SiteTree"."Sort"' work.
 		if(preg_match('/^"?(?<column>[^"\s]+)"?(\s+(?<direction>((asc)|(desc))(ending)?))?$/i', $column, $match)) {
```

Comment:
I'm interested why we need the quote removal above, when this regexp already detects quotes.

How does Table.Column sort syntax work on a list with no tables? I'm not able to see how this makes ArrayList sorting work where it didn't before. Maybe if there were some unit tests?